### ISSUE TITLE
Change Request: Add option to preserve logical grouping of member types

### ISSUE THREAD

**Issue by HakeemTheEmperor** [eslint/eslint#21283]
### What rule do you want to change?

sort-keys

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
const user = {
    id: 1,
    name: "Ada",
    age: 36,
    get fullName() {
        return `${this.name}`;
    }
};
```

### What does the rule currently do for this code?

The rule currently checks the keys alphabetically and does not distinguish between plain data properties and accessor properties.

As a result, it will likely report fullName as being out of order because it is a getter, and the rule treats it as just another key to sort lexicographically.

In practice, the rule may produce a warning like:

- fullName should be before id / name / age according to alphabetical ordering

This is stricter than many developers expect when they intentionally group data fields first and accessor methods after them.

### What will the rule do after it's changed?

After the change, if the user enables the new option, the rule should allow a logical grouping such as:

```
const user = {
    id: 1,
    name: "Ada",
    age: 36,
    get fullName() {
        return `${this.name}`;
    }
};
```

and still enforce ordering within each group.

For example:

- plain data properties sorted alphabetically within the data group
- accessors kept together after the data fields
- computed properties optionally grouped separately if the option allows it

The rule would still warn in the default mode when the object is not alphabetized, but users who choose the new grouping behavior would not get noisy warnings for a semantically grouped layout.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

This feels like a good candidate for a rule change because it does not introduce a brand-new rule; it improves the behavior of an existing one.

I think this should be:

- opt-in
- backward compatible
- documented carefully
- covered by regression tests for mixed member types

The key concern is avoiding a breaking change for current users while still making the rule more useful for real-world modern object patterns.

**Comment by DMartens** [comment]
The rule `sort-keys` is [frozen](https://eslint.org/docs/latest/contribute/core-rules#frozen-rules) and as such is not accepting feature requests.
You can fork the rule and can request via our help channels (e.g. discord) if you require help with implementing it.
